testnet/stacks-node/src/tests/neon_integrations.rs

```diff
@@ -2213,6 +2217,319 @@ fn stx_delegate_btc_integration_test() {
+    assert!(
+        btc_regtest_controller
+            .submit_operation(
+                StacksEpochId::Epoch25,
+                stack_stx_op_with_some_signer_key,
+                &mut spender_signer_1,
+                1
+            )
+            .is_some(),
+        "Stack STX operation with some signer key should submit successfully"
+    );
+
+    let stack_stx_op_with_no_signer_key = BlockstackOperationType::StackStx(StackStxOp {
+        sender: spender_stx_addr_2.clone(),
+        reward_addr: pox_addr.clone(),
+        stacked_ustx: 100000,
+        num_cycles: reward_cycle.try_into().unwrap(),
+        signer_key: None,
+        // to be filled in
+        vtxindex: 0,
+        txid: Txid([0u8; 32]),
+        block_height: 0,
+        burn_header_hash: BurnchainHeaderHash::zero(),
+    });
+
+    let mut spender_signer_2 = BurnchainOpSigner::new(signer_sk_2.clone(), false);
+    assert!(
+        btc_regtest_controller
+            .submit_operation(
+                StacksEpochId::Epoch25,
+                stack_stx_op_with_no_signer_key,
+                &mut spender_signer_2,
+                1
+            )
+            .is_some(),
+        "Stack STX operation with no signer key should submit successfully"
+    );
+
+    info!("Submitted 2 stack STX ops at height {block_height}, mining a few blocks...");
+
+    // the second block should process the vote, after which the balaces should be unchanged
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+
+    let mut stack_stx_found = false;
+    let blocks = test_observer::get_blocks();
+    info!("stack event observer num blocks: {:?}", blocks.len());
+    for block in blocks.iter() {
+        let transactions = block.get("transactions").unwrap().as_array().unwrap();
+        info!(
+            "stack event observer num transactions: {:?}",
+            transactions.len()
+        );
+        for tx in transactions.iter() {
+            let raw_tx = tx.get("raw_tx").unwrap().as_str().unwrap();
+            if raw_tx == "0x00" {
+                info!("Found a burn op: {:?}", tx);
+                let burnchain_op = tx.get("burnchain_op").unwrap().as_object().unwrap();
+                if !burnchain_op.contains_key("stack_stx") {
+                    warn!("Got unexpected burnchain op: {:?}", burnchain_op);
+                    panic!("unexpected btc transaction type");
+                }
+                let stack_stx_obj = burnchain_op.get("stack_stx").unwrap();
+                let signer_key_found = stack_stx_obj
+                    .get("signer_key")
+                    .expect("Expected signer_key in burn op")
+                    .as_str()
+                    .unwrap();
+                assert_eq!(signer_key_found, signer_key.to_hex());
+
+                stack_stx_found = true;
+            }
```

Comment:
Can you add some assertion that we only found 1 stack-stx op, and not two? Just so that we have some validation that the op without a signer key was not submitted?

I think in this case it could also be helpful to try and query `sortdb` to see that the op without a signer key was found (aka it was "valid") but still not executed.

Reply:
I think `sortdb` call helped discover an issue directly or undirectly....because the query returned 0.
I came across this log while grepping and it seems since I am not including a signature in `StackStxOp` and passing `None` when submitting a clarity_tx, It's failing because an authorization wasn't added for that signer-key (I suppose that happens via `set-signer-key-authorization`.
I'll see if I can fix that up.

> DEBG [1709861612.139892] [stackslib/src/chainstate/stacks/db/blocks.rs:4202] [miner-block-http://127.0.0.1:33451] StackStx burn op rejected by PoX contract., txid: 6110c53d30f52cfca7498aef8cdc45c7932a7cddf9d7dd97034108449f99e597, burn_block: 31c8df6a265df548aa8f72ae56c625307067ef96c94e9ac57e271b3c144e032e, contract_call_ecode: 19

Reply:
My [last commit] (https://github.com/stacks-network/stacks-core/pull/4412/commits/a0f57913ab208569b7d9a76d2ef5fa4b1a18477e) takes care of this as verified by the added assertions and logs..._but_ I am not sure if this is what we want? Since it was a big thing to miss and I'm not sure if there was a diff idea behind making this work without a signature _and_ without an additional Clarity call.

Logs showing both Clarity transactions going through:

> DEBG [1709878156.643139] [stackslib/src/chainstate/stacks/db/blocks.rs:4668] [chains-coordinator-127.0.0.1:39177] Processed set-signer-key-authorization

>DEBG [1709878156.663462] [stackslib/src/chainstate/stacks/db/blocks.rs:4230] [chains-coordinator-
127.0.0.1:39177] Processed StackStx burnchain op, amount_ustx: 10000000000000, num_cycles: 6, burn_block_height: 210, sender: ST3WM51TCWMJYGZS1QFMC28DH5YP86782YGR113C1, reward_addr: {"Standard":[{"version":26,"bytes":"f942874ce525e87f21bbe8c121b12fac831d02f4"},"SerializeP2PKH"]}, txid: cc5d575bd9cd01553943495cddb0a05d1a071d7a757841dd252474b54b3683ef

Reply:
More updates on this: Integ tests were passing with this change (the burn op transaction is found in block events and from sortdb)
But, after I tried merging with next and resolved conflicts, the burn op transaction was no longer being found in block events (but it was still found in sortdb)
I didn't push that up since it broke the test and above approach wasn't confirmed. 

I found below warning that I think is indicative of the issue:

> WARN [1710112824.503133] [stackslib/src/chainstate/stacks/db/blocks.rs:5610] [chains-coordinator-127.0.0.1:43685] Block 6b225be385245404ed8ddc3ecefee607b67d763799fc697ff575d702b21a4ae9 state root mismatch: expected 32969e8edfa086f9f8c60782c33f1a5be0feded5609e8d1380cb108b44e0fe95, got cf9705f9acb1397c4ee6cf4214c4b906c32f188ddebac92b0f298db7988200a9
WARN [1710112824.503727] [stackslib/src/chainstate/stacks/db/blocks.rs:6348] [chains-coordinator-127.0.0.1:43685] Encountered invalid block: Block 6b225be385245404ed8ddc3ecefee607b67d763799fc697ff575d702b21a4ae9 state root mismatch: expected 32969e8edfa086f9f8c60782c33f1a5be0feded5609e8d1380cb108b44e0fe95, got cf9705f9acb1397c4ee6cf4214c4b906c32f188ddebac92b0f298db7988200a9

I suspect this is happening because the `set-signer-key-authorization` is not properly updating the `StacksTransactionReceipt` as in the current change I have, I didn't add a a new `BlockstackOperationType` and just called with without adding a `receipt`.
Not sure why this is passing without bringing in changes in `next`
But, if we are to make this additional `set-signer-key-authorization` clarity call for every `stack-stx` call, then we might need to define a new `BlockstackOperationType` for it.

Reply:
I suspect this is failing because `max-amount` and `auth-id` were merged into next, so the clarity transaction is now failing. This updates both `stack-stx` and `set-signer-key-authorization`. 

The fix would be to append both of those params to those functions (the "vanilla" tx and the burn op). The burn op will need to be updated to include those two params in the data field.

Let me know if you need pointers about how to use those fields in tests, but it's pretty straightforward:

- You can use `u128::MAX` for `max-amount`
- Use any number for auth ID, they just have to match

Apologies for not catching this earlier, I should have thought about how the test would fail without the prior authorization.

Also, I wouldn't worry about making a burn op for `set-signer-key-authorization` in this PR

Reply:
This is done in my recent commit now and it helped the Neon integ test to pass locally. But the Nakamoto integ test fails, and it seems to be a the same issue with [vote burn op](https://github.com/stacks-network/stacks-core/actions/runs/8238927057/job/22531584884?pr=4376#step:5:2416) but I wasn't able to figure out why.